### PR TITLE
docs: Migration guide for 2.0

### LINKED ISSUE #245: Migration guide for 2.0
Compile a list of all changes for `2.0` milestone.

Go through each PR to note implementation details!

### PR DESCRIPTION
**Pull request by peterkos**
Fixes #245 

Also renamed `maintenance.md` -> `updating-hm.md` to match its article title.

Screen Shot 2020-09-21 at 12 21 34 AM

### CODE REVIEW

File: app/controllers/rsvps_controller.rb [hunk 1]
```diff
@@ -58,6 +62,17 @@ def update
     update_acc_status
     update_bus_list
 
+    bus_after_rsvp = @questionnaire.bus_list_id
+    acc_status_after_rsvp = @questionnaire.acc_status
+
+    if bus != nil && (acc_status != acc_status_after_rsvp || bus != bus_after_rsvp) && @questionnaire.is_bus_captain == true
```
[the remark this answers is not in the document]
Comment (peterkos) [reply app/controllers/rsvps_controller.rb]: fixed by backmerge

File: docs/2.0-migration-guide.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+---
+id: 2.0-migration-guide
+title: 2.0 Migration Guide
```
Comment (cbaudouinjr) [inline docs/2.0-migration-guide.md:3]: We may want to change the name to something like "What's new in 2.0" as this doesn't come off as a guide, they're more like update docs!

File: docs/2.0-migration-guide.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+HackathonManager 2.0 has a lot of new features, changes, and bug fixes! This page serves to detail the updates that may affect your existing instance of HackathonManager.
+
+
+## Updating
+
+If you are not currently on v2.0, see the guide for [updating HM](updating-hm.md).
+
+
+## New Features
```
Comment (cbaudouinjr) [inline docs/2.0-migration-guide.md:14]: Don't forget the RSVP reminder email!

**Review by cbaudouinjr — CHANGES_REQUESTED** [review]

File: docs/2.0-migration-guide.md (hunk 2 above)
Comment (peterkos) [reply docs/2.0-migration-guide.md:3]: I see!

**Review by cbaudouinjr — APPROVED** [review]

**Pull request merged**